store/src/test/java/com/dropbox/android/external/store4/impl/FlowTestExt.kt

```diff
@@ -15,19 +15,56 @@
  */
 package com.dropbox.android.external.store4.impl
 
+import com.google.common.truth.FailureMetadata
+import com.google.common.truth.Subject
+import com.google.common.truth.Truth
 import com.google.common.truth.Truth.assertThat
 import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.take
-import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.TestCoroutineScope
 
-/**
- * Asserts only the [expected] items by just taking that many from the stream
- *
- * Use this when Pipeline has an infinite part (e.g. Persister or a never ending fetcher)
- */
-@ExperimentalCoroutinesApi
-suspend fun <T> Flow<T>.assertItems(vararg expected: T) {
-    assertThat(this.take(expected.size).toList())
-            .isEqualTo(expected.toList())
+@UseExperimental(ExperimentalCoroutinesApi::class)
+internal fun <T> TestCoroutineScope.assertThat(flow: Flow<T>): FlowSubject<T> {
+    return Truth.assertAbout(FlowSubject.Factory<T>(this)).that(flow)
+}
+
+@UseExperimental(ExperimentalCoroutinesApi::class)
+internal class FlowSubject<T> constructor(
+    failureMetadata: FailureMetadata,
+    private val testCoroutineScope: TestCoroutineScope,
+    private val actual: Flow<T>
+) : Subject(failureMetadata, actual) {
+    /**
+     * Takes all items in the flow that are available by collecting on it as long as there are
+     * active jobs in the given [TestCoroutineScope].
+     *
+     * It ensures all expected items are dispatched as well as no additional unexpected items are
+     * dispatched.
+     */
+    suspend fun emitsExactly(vararg expected: T) {
+        val collectedSoFar = mutableListOf<T>()
+        val collectJob = testCoroutineScope.launch {
+            actual.collect {
+                collectedSoFar.add(it)
+                assertThat(collectedSoFar.size).isAtMost(expected.size)
+            }
+        }
+        testCoroutineScope.advanceUntilIdle()
+        collectJob.cancel()
+        assertThat(collectedSoFar).isEqualTo(expected.toList())
```

Comment:
while slightly better (as it will show the actual collection) maybe we can put a better error message here too (e.g "expected emissions: $expected, but was $actual")

Reply:
added a message for the expectation but not the values since Truth already exports those values for us.